File: java/src/main/java/org/github/arosecra/book/pipeline/model/Pipeline.java
package org.github.arosecra.book.pipeline.model;

import java.util.ArrayList;
import java.util.List;

public class Pipeline {
	public List<JobStep> steps = new ArrayList<>();
	public String name;
	public String uses;
	public String produces;
	public String remoteDirectory;
	
	public Pipeline addStep(JobStep step) {
		this.steps.add(step);
		return this;
	}
	
	public Pipeline setName(String name) {
		this.name = name;
		return this;
	}
	
	public Pipeline setRemoteDirectory(String remoteDirectory) {
		this.remoteDirectory = remoteDirectory;
		return this;
	}
	
	public Pipeline setUses(String uses) {
		this.uses = uses;
		return this;
	}
	
	public Pipeline setProduces(String produces) {
		this.produces = produces;
		return this;
	}
	
}

File: ts-pipeline/src/model/pipeline.ts
import { JobStep } from "./job-step";


export class Pipeline {

	public steps: JobStep[] = [];
	public name: string;
	public uses: string[] = [];
	public produces: string;
	public propertyCheck: (file: string) => boolean;
	public remoteDirectory: string;
	
	public addStep(step: JobStep): Pipeline {
		this.steps.push(step);
		return this;
	}
	
	public setName(name: string) {
		this.name = name;
		return this;
	}
	
	public setRemoteDirectory(remoteDirectory: string): Pipeline {
		this.remoteDirectory = remoteDirectory;
		return this;
	}
	
	public setUses(uses: string[]): Pipeline {
		this.uses.push(...uses);
		return this;
	}
	
	public setProduces(produces: string): Pipeline {
		this.produces = produces;
		return this;
	}

	public setPropertyCheck(propertyCheck: (file: string) => boolean): Pipeline {
		this.propertyCheck = propertyCheck;
		return this;
	}

}